benchexec/tools/nitwit.py

```diff
@@ -50,24 +43,24 @@ def name(self):
     def cmdline(self, executable, options, task, rlimits):
         if task.property_file:
             options = options + ["-p", task.property_file]
-        return [executable] + options + list(task.input_files_or_identifier)
+        return [executable] + options + task.single_input_file
 
     def determine_result(self, run):
         """
         See README.md at https://github.com/moves-rwth/nitwit-validator for information
         about result codes.
         @return: status of validator after executing a run
         """
-        if run.exit_code.signal is None and (run.exit_code.value == 0 or run.exit_code.value == 245):
+        if run.exit_code.value in [0, 245]:
             status = result.RESULT_FALSE_REACH
-        elif run.exit_code.value is None or run.exit_code.value in [-9, 9]:
+        elif (run.exit_code.signal == 9) or run.exit_code.value == 9:
```

Comment:
As you said in your comment that Nitwit never returns exit code 9, the second part seems unnecessary. If the process is killed by a signal, `value` won't be set.

But a check for "SIGKILL => TIMEOUT" is not necessary at all in a tool-info module. If there is a timeout, BenchExec will mark it as such automatically. In fact, the current code will only hide other crashes with signal 9 behind a `TIMEOUT` status, which I guess is not intended. So I would suggest to remove this branch completely.

Reply:
Removed.